src/lib.rs

```diff
@@ -1674,6 +1674,64 @@ impl<B: BitBlock> BitVec<B> {
             carry = curr_carry;
         }
     }
+
+    /// Appends an element if there is sufficient spare capacity, otherwise an error is returned
+    /// with the element.
+    ///
+    /// Unlike [`push`] this method will not reallocate when there's insufficient capacity.
+    /// The caller should use [`reserve`] to ensure that there is enough capacity.
+    ///
+    /// [`push`]: Self::push
+    /// [`reserve`]: Self::reserve
+    ///
+    /// # Examples
+    /// ```
+    /// use bit_vec::BitVec;
+    ///
+    /// let initial_capacity = 64;
+    /// let mut bitvec = BitVec::with_capacity(64);
+    ///
+    /// for _ in 0..initial_capacity - 1 {
+    ///     bitvec.push(false);
+    /// }
+    ///
+    /// assert_eq!(bitvec.len(), initial_capacity - 1); // there is space for only 1 bit
+    ///
+    /// assert_eq!(bitvec.push_within_capacity(true), Ok(())); // Successfully push a bit
+    /// assert_eq!(bitvec.len(), initial_capacity); // So we can't push within capacity anymore
+    ///
+    /// assert_eq!(bitvec.len(), initial_capacity);
```

Comment:
I think you accidentally added or have eaten a line here

Reply:
It was intentional, actually. I wanted to "group" similar asserts together

Reply:
> It was intentional, actually. I wanted to "group" similar asserts together

But the order of asserts is wrong, will fix that